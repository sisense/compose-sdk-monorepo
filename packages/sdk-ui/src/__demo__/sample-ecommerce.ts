import { measureFactory } from '@sisense/sdk-data';

import { Commerce, Brand, Country, Category, DataSource } from './sample-ecommerce-autogenerated';

export { Commerce, Brand, Category, Country, DataSource };

/** adding predefined set of measures */
export const Measures = {
  SumCost: measureFactory.sum(Commerce.Cost).format('0,0$'),
  AvgCost: measureFactory.average(Commerce.Cost).format('0,0$'),
  SumRevenue: measureFactory.sum(Commerce.Revenue, 'Total Revenue').format('0,0$'),
  Quantity: measureFactory.sum(Commerce.Quantity, 'Total Quantity').format('0,0'),
  CountBrands: measureFactory.countDistinct(Commerce.BrandID).format('0,0'),
};
