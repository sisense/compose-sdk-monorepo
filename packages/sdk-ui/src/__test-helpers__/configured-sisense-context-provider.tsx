import { mockToken, mockUrl } from '@/__mocks__/msw';
import { SisenseContextProviderProps } from '@/props';
import { SisenseContextProvider } from '@/sisense-context/sisense-context-provider';

const contextProviderProps: SisenseContextProviderProps = {
  url: mockUrl,
  token: mockToken,
  defaultDataSource: 'Sample ECommerce',
  appConfig: {
    queryCacheConfig: { enabled: false },
    trackingConfig: { enabled: false },
  },
};

export const ConfiguredSisenseContextProvider: React.FC<{
  children: React.ReactNode;
}> = ({ children }) => {
  return <SisenseContextProvider {...contextProviderProps}>{children}</SisenseContextProvider>;
};
