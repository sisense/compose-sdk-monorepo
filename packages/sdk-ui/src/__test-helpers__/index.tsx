/**
 * Common test utilities
 */
import { PropsWithChildren, ReactElement } from 'react';
import { render, RenderResult } from '@testing-library/react';
import type { Cell, Data } from '@ethings-os/sdk-data';
import isObject from 'lodash-es/isObject';
import userEvent from '@testing-library/user-event';
import { SisenseContext, SisenseContextPayload } from '../sisense-context/sisense-context';
import { Authenticator, HttpClient } from '@ethings-os/sdk-rest-client';
import { ClientApplication } from '../app/client-application';
import { DeepPartial } from 'ts-essentials';
import { useTranslation } from 'react-i18next';
import { translation } from '../translation/resources/en';
import { CustomWidgetsProvider } from '../custom-widgets-provider';
import { MenuProvider } from '@/common/components/menu/menu-provider';

type UserSetupFn = (typeof userEvent)['setup'];
interface SetupResult extends RenderResult {
  user: ReturnType<UserSetupFn>;
}
/**
 * This helper method combines the userEvent.setup() and render() calls at the
 * start of a unit test using @testing-library/react.
 *
 * Based on recommendation by: https://testing-library.com/docs/user-event/intro/#writing-tests-with-userevent
 *
 * @param jsx - React component instance to test, passed directly to render()
 * @param useFakeTimers - true if Vitest fake timers are enabled
 * @returns
 */
export function setup(jsx: ReactElement, useFakeTimers = false): SetupResult {
  return {
    // Need to set "advanceTimers" if we're using Vitest's fake timers.
    // The solution in https://github.com/testing-library/user-event/issues/833#issuecomment-1171452841
    // is for Jest, but this seems to work for Vitest as well.
    user: userEvent.setup(
      useFakeTimers ? { advanceTimers: (ms) => vi.advanceTimersByTime(ms) } : undefined,
    ),
    ...render(jsx),
  };
}

export const MockedHighchartsReact = ({
  options,
}: {
  /**
   * Highcharts chart configuration object.
   * Please refer to the Highcharts (API documentation)[https://api.highcharts.com/highcharts/].
   */
  options: object;
}) => {
  return <div>{JSON.stringify(options)}</div>;
};

type MockedSisenseContextProviderProps = PropsWithChildren<
  DeepPartial<Omit<SisenseContextPayload, 'app'>>
>;

const mockHttpClient = new HttpClient(
  'http://mock-url/sometenant?someparam=true',
  {} as Authenticator,
  'mock-env',
);
export const MockedSisenseContextProvider = ({
  children,
  tracking = {
    enabled: true,
  },
  isInitialized = true,
  errorBoundary,
}: MockedSisenseContextProviderProps) => {
  return (
    <SisenseContext.Provider
      value={{
        app: {
          httpClient: mockHttpClient,
        } as ClientApplication,
        tracking: {
          enabled: tracking.enabled ?? true,
          packageName: tracking.packageName ?? 'sdk-ui',
        },
        isInitialized,
        errorBoundary: { showErrorBox: true, ...errorBoundary },
      }}
    >
      <MenuProvider>
        <CustomWidgetsProvider>{children}</CustomWidgetsProvider>
      </MenuProvider>
    </SisenseContext.Provider>
  );
};

/**
 * Extends the provided data by blurring specified rows.
 *
 * @param {Data} data - The original data object.
 * @param {number[]} blurRowsIndexes - An array of row indexes to be blurred.
 * @returns {Data} The data object with blurred rows.
 */
export const withBlurredRows = (data: Data, blurRowsIndexes: number[]) => ({
  ...data,
  rows: data.rows.map((row, rowIndex) =>
    row.map((value) => {
      const cell: Cell = isObject(value) ? value : { data: value };

      if (blurRowsIndexes.includes(rowIndex)) {
        return { ...cell, blur: true };
      }
      return cell;
    }),
  ),
});

/**
 * Mocking the i18next translation functionality for tests.
 */
export const setupI18nMock = () => {
  vi.mock('react-i18next', async () => {
    const original: { useTranslation: typeof useTranslation } = await vi.importActual(
      'react-i18next',
    );

    // Helper function to get nested translations safely
    const getNestedTranslation = (key: string): string | undefined => {
      return key.split('.').reduce((accumulator: any, currentKey: string) => {
        if (accumulator && typeof accumulator === 'object' && currentKey in accumulator) {
          return accumulator[currentKey];
        }
        return undefined;
      }, translation) as string | undefined;
    };

    return {
      ...original,
      useTranslation: () => ({
        ...original.useTranslation(),
        t: vi.fn((key, options?: Record<string, any>) => {
          const translatedValue = getNestedTranslation(key);
          if (translatedValue && options) {
            // Replace variables in the translated string
            return translatedValue.replace(/{{(.*?)}}/g, (_, varName) => {
              const trimmedVarName = varName.trim();
              return options[trimmedVarName] || `{{${trimmedVarName}}}`; // Fallback if variable is not found
            });
          }
          return translatedValue ?? key; // Fallback to the key if translation is not found
        }),
      }),
    };
  });
};
