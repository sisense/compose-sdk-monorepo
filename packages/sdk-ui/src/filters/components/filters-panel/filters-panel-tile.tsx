import {
  DataSource,
  CascadingFilter,
  CustomFilter,
  DateRangeFilter,
  Filter,
  LevelAttribute,
  MeasureFilter,
  MembersFilter,
  NumericFilter,
  RankingFilter,
  RelativeDateFilter,
  TextFilter,
} from '@sisense/sdk-data';
import { MemberFilterTile } from '../member-filter-tile';
import { CriteriaFilterTile } from '../criteria-filter-tile';
import { DateRangeFilterTile, RelativeDateFilterTile } from '../date-filter';
import { FilterTile } from '../filter-tile';
import { useTranslation } from 'react-i18next';
import { CascadingFilterTile } from '../cascading-filter-tile';
import { CustomFilterTile } from '../custom-filter-tile';

/**
 * Props of the {@link FiltersPanelTile} component
 *
 * @internal
 */
export type FiltersPanelTileProps = {
  /** Filter to display */
  filter: Filter;
  /** Callback to handle filter change */
  onChange: (filter: Filter | null) => void;
  /** Default data source used for filter tiles */
  defaultDataSource?: DataSource;
};

/**
 * Filter tile component that renders a filter based on its type
 *
 * @internal
 */
export const FiltersPanelTile = ({
  filter,
  onChange,
  defaultDataSource,
}: FiltersPanelTileProps) => {
  const { t } = useTranslation();

  const attribute = filter.attribute;
  const title = attribute.name;
  const props = {
    attribute,
    title,
    onChange,
    onUpdate: onChange,
    ...(defaultDataSource ? { dataSource: defaultDataSource } : null),
  };
  // checking for custom filters first to prevent conversion attempts
  if (filter instanceof CustomFilter) {
    return <CustomFilterTile {...props} filter={filter} />;
  }

  if (filter instanceof MembersFilter) {
    return <MemberFilterTile {...props} filter={filter} />;
  }

  if (filter instanceof DateRangeFilter) {
    return (
      <DateRangeFilterTile
        {...props}
        filter={filter}
        attribute={attribute as LevelAttribute}
        tiled
      />
    );
  }

  if (filter instanceof RelativeDateFilter) {
    return <RelativeDateFilterTile {...props} filter={filter} arrangement="vertical" />;
  }

  if (
    filter instanceof MeasureFilter ||
    filter instanceof NumericFilter ||
    filter instanceof TextFilter ||
    filter instanceof RankingFilter
  ) {
    return <CriteriaFilterTile {...props} filter={filter} />;
  }

  if (filter instanceof CascadingFilter) {
    return <CascadingFilterTile filter={filter} onChange={onChange} />;
  }

  return (
    <FilterTile
      title={t('unsupportedFilter.title')}
      renderContent={() => (
        <p className="csdk-text-center csdk-text-[13px]">{t('unsupportedFilter.message')}</p>
      )}
      design={{ header: { isCollapsible: false } }}
    />
  );
};
