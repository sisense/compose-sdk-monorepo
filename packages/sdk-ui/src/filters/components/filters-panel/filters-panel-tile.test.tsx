/** @vitest-environment jsdom */
import { findByLabelText, findByText, render } from '@testing-library/react';
import {
  DateRangeFilter,
  RelativeDateFilter,
  createAttribute,
  LevelAttribute,
  filterFactory,
  measureFactory,
  Filter,
} from '@sisense/sdk-data';
import { vi } from 'vitest';
import { FiltersPanelTile } from './filters-panel-tile';
import { MockedSisenseContextProvider } from '@/__test-helpers__';
import * as DM from '@/__test-helpers__/sample-ecommerce';

const mockOnChange = vi.fn();

const mockAttribute = createAttribute({
  name: 'Test Revenue',
  type: 'numeric-attribute',
  expression: '[Commerce.Revenue]',
});

describe('FiltersPanelTile', () => {
  // tries to execute query
  // TODO: mock the query execution or redesign the test
  it.skip('renders MemberFilterTile for MembersFilter', async () => {
    const mockFilter = filterFactory.members(mockAttribute, ['1', '2', '3']);
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByLabelText(container, mockAttribute.name);
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  // tries to execute query
  // TODO: mock the query execution or redesign the test
  it.skip('renders DateRangeFilterTile for DateRangeFilter', async () => {
    const mockFilter = new DateRangeFilter(
      mockAttribute as LevelAttribute,
      '2021-01-01',
      '2021-12-31',
    );
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByText(container, mockAttribute.name);
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('renders RelativeDateFilterTile for RelativeDateFilter', async () => {
    const mockFilter = new RelativeDateFilter(DM.Commerce.Date.Years, 0, 1);
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByText(container, DM.Commerce.Date.Years.name);
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('renders locked RelativeDateFilter component', async () => {
    const mockFilter = new RelativeDateFilter(DM.Commerce.Date.Years, 0, 1) as Filter;
    mockFilter.locked = true;
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByText(container, DM.Commerce.Date.Years.name);
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  // throws 'Cannot convert object to primitive value'
  // TODO: investigate the error and fix the test
  it.skip('renders CriteriaFilterTile for MeasureFilter', async () => {
    const mockFilter = filterFactory.measureEquals(measureFactory.sum(mockAttribute), 100);
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByLabelText(container, mockAttribute.name);
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('renders CriteriaFilterTile for NumericFilter', async () => {
    const mockFilter = filterFactory.greaterThan(mockAttribute, 100);
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByText(container, mockAttribute.name);
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('renders locked CriteriaFilterTile', async () => {
    const mockFilter = filterFactory.greaterThan(mockAttribute, 100);
    mockFilter.locked = true;
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByText(container, mockAttribute.name);
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('renders CriteriaFilterTile for TextFilter', async () => {
    const mockFilter = filterFactory.doesntStartWith(mockAttribute, '100');
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByText(container, mockAttribute.name);
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('renders CriteriaFilterTile for RankingFilter', async () => {
    const mockFilter = filterFactory.topRanking(
      mockAttribute,
      measureFactory.average(mockAttribute),
      10,
    );
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByText(container, mockAttribute.name);
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('renders CustomFilterTile for advnced filter type', async () => {
    const mockFilter = filterFactory.customFilter(mockAttribute, {});
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByText(container, mockAttribute.name);
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });

  it('renders FilterTile for unsupported filter type', async () => {
    const mockFilter = { attribute: { name: 'Unsupported Filter' } } as Filter;
    const { container } = render(
      <MockedSisenseContextProvider>
        <FiltersPanelTile filter={mockFilter} onChange={mockOnChange} />
      </MockedSisenseContextProvider>,
    );
    const tile = await findByText(container, 'unsupportedFilterMessage');
    expect(tile).toBeTruthy();
    expect(container).toMatchSnapshot();
  });
});
