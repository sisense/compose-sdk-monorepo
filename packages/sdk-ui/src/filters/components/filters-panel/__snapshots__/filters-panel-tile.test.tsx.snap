// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FiltersPanelTile > renders CriteriaFilterTile for NumericFilter 1`] = `
<div>
  <div
    class="css-xhbwdz"
  >
    <div
      class="css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255);"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <svg
          aria-label="arrow-down"
          class="csdk-transition csdk-ml-[4px] csdk-cursor-pointer -csdk-rotate-90"
          fill="#5B6372"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.589 10L7 10.777 11.99 15 17 10.777 16.413 10l-4.422 3.727z"
            fill="#5B6372"
            fill-rule="nonzero"
          />
        </svg>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114);"
        >
          TestRevenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="csdk-p-[4px] csdk-leading-[26px] csdk-mx-auto csdk-text-[13px] csdk-whitespace-nowrap csdk-flex csdk-flex-wrap csdk-gap-x-1 csdk-justify-center"
        >
          criteriaFilter.greaterThan
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-17lmbjk css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FiltersPanelTile > renders CriteriaFilterTile for RankingFilter 1`] = `
<div>
  <div
    class="css-xhbwdz"
  >
    <div
      class="css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255);"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <svg
          aria-label="arrow-down"
          class="csdk-transition csdk-ml-[4px] csdk-cursor-pointer -csdk-rotate-90"
          fill="#5B6372"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.589 10L7 10.777 11.99 15 17 10.777 16.413 10l-4.422 3.727z"
            fill="#5B6372"
            fill-rule="nonzero"
          />
        </svg>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114);"
        >
          TestRevenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="csdk-p-[4px] csdk-leading-[26px] csdk-mx-auto csdk-text-[13px] csdk-whitespace-nowrap csdk-flex csdk-flex-wrap csdk-gap-x-1 csdk-justify-center"
        >
          criteriaFilter.top
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-17lmbjk css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FiltersPanelTile > renders CriteriaFilterTile for TextFilter 1`] = `
<div>
  <div
    class="css-xhbwdz"
  >
    <div
      class="css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255);"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <svg
          aria-label="arrow-down"
          class="csdk-transition csdk-ml-[4px] csdk-cursor-pointer -csdk-rotate-90"
          fill="#5B6372"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.589 10L7 10.777 11.99 15 17 10.777 16.413 10l-4.422 3.727z"
            fill="#5B6372"
            fill-rule="nonzero"
          />
        </svg>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114);"
        >
          TestRevenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="csdk-p-[4px] csdk-leading-[26px] csdk-mx-auto csdk-text-[13px] csdk-whitespace-nowrap csdk-flex csdk-flex-wrap csdk-gap-x-1 csdk-justify-center"
        >
          criteriaFilter.notStartsWith
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-17lmbjk css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FiltersPanelTile > renders CustomFilterTile for advnced filter type 1`] = `
<div>
  <div
    class="css-xhbwdz"
  >
    <div
      class="css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255);"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114);"
        >
          TestRevenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <p
          class="csdk-text-center csdk-text-[13px]"
        >
          customFilterTileMessage
        </p>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-17lmbjk css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FiltersPanelTile > renders FilterTile for unsupported filter type 1`] = `
<div>
  <div
    class="css-xhbwdz"
  >
    <div
      class="css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255);"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114);"
        >
          unsupportedFilter.title
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="csdk-p-[8px] csdk-text-center csdk-text-[13px]"
        >
          <span>
            unsupportedFilter.message
          </span>
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-17lmbjk css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FiltersPanelTile > renders RelativeDateFilterTile for RelativeDateFilter 1`] = `
<div>
  <div
    class="css-xhbwdz"
  >
    <div
      class="css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255);"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <svg
          aria-label="arrow-down"
          class="csdk-transition csdk-ml-[4px] csdk-cursor-pointer -csdk-rotate-90"
          fill="#5B6372"
          height="16"
          viewBox="0 0 24 24"
          width="16"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7.589 10L7 10.777 11.99 15 17 10.777 16.413 10l-4.422 3.727z"
            fill="#5B6372"
            fill-rule="nonzero"
          />
        </svg>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114);"
        >
          TestRevenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="csdk-leading-[26px] csdk-mx-auto csdk-my-2 csdk-px-1 csdk-text-[13px] csdk-whitespace-nowrap csdk-flex csdk-flex-wrap csdk-gap-x-1 csdk-justify-center"
        >
          dateFilter.next 1 undefined dateFilter.from dateFilter.today
        </div>
      </main>
      <footer
        class="css-1wpi5bx"
      >
        <span
          class="MuiSwitch-root MuiSwitch-sizeSmall css-17lmbjk css-1diaq70-MuiSwitch-root"
        >
          <span
            class="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorPrimary Mui-checked Mui-checked css-byenzh-MuiButtonBase-root-MuiSwitch-switchBase"
          >
            <input
              checked=""
              class="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
              name="tile-switch"
              role="switch"
              type="checkbox"
            />
            <span
              class="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
            />
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </span>
          <span
            class="MuiSwitch-track css-1yjjitx-MuiSwitch-track"
          />
        </span>
      </footer>
    </div>
  </div>
</div>
`;

exports[`FiltersPanelTile > renders locked CriteriaFilterTile 1`] = `
<div>
  <div
    class="css-xhbwdz"
  >
    <div
      class="css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255);"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <svg
          fill="#b4b4b4"
          height="22"
          viewBox="0 0 22 22"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(5.000000,5.000000) scale(.70, 0.6)"
          >
            <path
              d="m3,9v11h14V9M4,9V6c0-3.3 2.7-6 6-6c3.3,0 6,2.7 6,6v3H14V6c0-2.2-1.8-4-4-4-2.2,0-4,1.8-4,4v3"
            />
            <rect
              fill="#dcdcdc"
              height="9"
              width="12"
              x="4"
              y="10"
            />
          </g>
        </svg>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114);"
        >
          TestRevenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="csdk-p-[4px] csdk-leading-[26px] csdk-mx-auto csdk-text-[13px] csdk-whitespace-nowrap csdk-flex csdk-flex-wrap csdk-gap-x-1 csdk-justify-center"
        >
          criteriaFilter.greaterThan
        </div>
        <div
          style="position: absolute; top: 4px; left: 4px; right: 4px; bottom: 4px; background-color: white; opacity: 0.5;"
        />
      </main>
      <footer
        class="css-1wpi5bx"
      />
    </div>
  </div>
</div>
`;

exports[`FiltersPanelTile > renders locked RelativeDateFilter component 1`] = `
<div>
  <div
    class="css-xhbwdz"
  >
    <div
      class="css-l4aa93"
      style="min-width: 215px; background-color: rgb(255, 255, 255);"
    >
      <header
        class="css-3aw1ep"
        style="color: rgb(91, 99, 114);"
      >
        <svg
          fill="#b4b4b4"
          height="22"
          viewBox="0 0 22 22"
          width="22"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            transform="translate(5.000000,5.000000) scale(.70, 0.6)"
          >
            <path
              d="m3,9v11h14V9M4,9V6c0-3.3 2.7-6 6-6c3.3,0 6,2.7 6,6v3H14V6c0-2.2-1.8-4-4-4-2.2,0-4,1.8-4,4v3"
            />
            <rect
              fill="#dcdcdc"
              height="9"
              width="12"
              x="4"
              y="10"
            />
          </g>
        </svg>
        <span
          class="csdk-text-[13px] csdk-mt-[6px] csdk-mb-[4px] csdk-ml-[7px] csdk-leading-[16px]"
          style="color: rgb(91, 99, 114);"
        >
          TestRevenue
        </span>
      </header>
      <main
        style="color: rgb(91, 99, 114); position: relative;"
      >
        <div
          class="csdk-leading-[26px] csdk-mx-auto csdk-my-2 csdk-px-1 csdk-text-[13px] csdk-whitespace-nowrap csdk-flex csdk-flex-wrap csdk-gap-x-1 csdk-justify-center"
        >
          dateFilter.next 1 undefined dateFilter.from dateFilter.today
        </div>
        <div
          style="position: absolute; top: 4px; left: 4px; right: 4px; bottom: 4px; background-color: white; opacity: 0.5;"
        />
      </main>
      <footer
        class="css-1wpi5bx"
      />
    </div>
  </div>
</div>
`;
